platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ConfigurationFilesPoller.java

```diff
@@ -80,16 +72,7 @@ public void init() throws IOException {
                 this.getClass()
                         .getName());
 
-        try {
-            LOGGER.debug("Registering path [{}] with Watch Service.",
-                    configurationDirectoryPath.toString());
-            configurationDirectoryPath.register(watchService, ENTRY_CREATE);
-        } catch (IOException e) {
-            LOGGER.error("Unable to register path [{}] with Watch Service",
-                    configurationDirectoryPath.toString(),
-                    e);
-            throw e;
-        }
+        watchService.addListener(new ListenerImpl());
```

Comment:
Actually, I wonder if we shouldn't just move the `addListener` call and `executorService.execute()` call inside the `register()` method and use the listener's `onStart` method to process existing files. This would guarantee that the processing of existing files and the notifications are all done on the same thread and should eliminate the risk of having a file being processed twice.